GitUI/UserControls/RevisionGrid/RevisionGridControl.cs

```diff
@@ -1537,6 +1506,82 @@ static IEnumerable<ObjectId> TryGetParents(IGitModule module, FilterInfo filterI
             }
         }
 
+        /// <summary>
+        /// Returns the name of a file in a specific revision (following renames and merge commits)
+        /// </summary>
+        public string? GetRevisionFileName(string path, ObjectId? objectId)
+        {
+            if (objectId is null)
+            {
+                return null;
+            }
+
+            if (FilePathByObjectId?.TryGetValue(objectId, out string? fileName) is true)
+            {
+                return fileName;
+            }
+
+            const string startOfObjectId = "????";
+
+            GitArgumentBuilder args = new("log")
+            {
+                // --name-only will list each filename on a separate line, ending with an empty line
+                // Find start of a new commit with a sequence impossible in a filename
+                $"--format=\"{startOfObjectId}%H\"",
+                "--name-only",
+                "--follow",
+                "--diff-merges=separate",
+                FindRenamesAndCopiesOpts(),
+                $"{objectId}^1..{objectId}",
+                "--",
+                path,
+            };
+
+            return ParseFileNames(args, startOfObjectId).FirstOrDefault(i => i.revision.Equals(objectId)).fileName;
+        }
+
+        private IEnumerable<(ObjectId revision, string fileName)> ParseFileNames(GitArgumentBuilder args, string startOfObjectId)
+        {
+            ExecutionResult result = Module.GitExecutable.Execute(args, outputEncoding: GitModule.LosslessEncoding, throwOnErrorExit: false);
+            LazyStringSplit lines = result.StandardOutput.LazySplit('\n');
+
+            // TODO Check the exit code and warn the user that rename detection could not be done.
```

Comment:
What are the plans for this TODO?

Reply:
I don't really know what are the "plans" - I just extracted the existing code to a dedicated method. But IMO if the git command did not execute successfully, the only thing we could do is just return early from this method (i. e. not return any file names).